File: demo-android/app/src/test/java/com/example/demo/pact/TasksGetIsValidPactTest.kt
package com.example.demo.pact

import au.com.dius.pact.consumer.MockServer
import au.com.dius.pact.consumer.dsl.PactBuilder
import au.com.dius.pact.consumer.junit5.PactConsumerTestExt
import au.com.dius.pact.consumer.junit5.PactTestFor
import au.com.dius.pact.core.model.V4Pact
import au.com.dius.pact.core.model.annotations.Pact
import kotlinx.coroutines.runBlocking
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith

@ExtendWith(PactConsumerTestExt::class)
@PactTestFor(providerName = "demo-service-tasks-get-is-valid")
class TasksGetIsValidPactTest {

    @Pact(consumer = PactFixtures.CONSUMER)
    fun getIsValidPact(builder: PactBuilder): V4Pact = builder.legacyHttpPact {
        given("validation result is true for the task")
            .uponReceiving("a request for task validation status")
            .pathFromProviderState(
                "/v1/tasks/isValid/\${taskId}",
                "/v1/tasks/isValid/${PactFixtures.TASK_ID}"
            )
            .method("GET")
            .willRespondWith()
            .status(200)
            .headers(mapOf("Content-Type" to "application/json"))
            .body("true")
    }

    @Test
    @PactTestFor(pactMethod = "getIsValidPact")
    fun shouldHaveGetIsValidContractWhenRequestingValidationStatus(mockServer: MockServer) {
        runBlocking {
            // Given
            val api = PactFixtures.createTaskApi(mockServer)

            // When
            api.isValid(PactFixtures.TASK_ID)
        }
    }
}

File: demo-react-native/src/test/pact/tasks.get-is-valid.pact.test.ts
import {MatchersV3} from '@pact-foundation/pact';

import {createTaskApi} from '@/data/remote/taskApi';
import {createPact} from './tasks.pact.fixtures';

const { boolean, fromProviderState } = MatchersV3;

const pact = createPact('demo-service-tasks-get-is-valid');
const taskId = '507f1f77bcf86cd799439011';
const isValidPath = fromProviderState(
  '/v1/tasks/isValid/${taskId}',
  `/v1/tasks/isValid/${taskId}`,
);

describe('tasks GET /v1/tasks/isValid/{id} pact', () => {
  it('should have get is valid contract', async () => {
    await pact
      .addInteraction()
      .given('validation result is true for the task')
      .uponReceiving('a request for task validation status')
      .withRequest('GET', isValidPath, (req) => {
        req.headers({ 'Content-Type': 'application/json' });
      })
      .willRespondWith(200, (res) => {
        res.headers({ 'Content-Type': 'application/json' });
        res.jsonBody(boolean(true));
      })
      .executeTest(async (mockServer) => {
        const api = createTaskApi(`${mockServer.url}/v1/`);
        await api.isValid(taskId);
      });
  });
});
